### PR TITLE
Xenobio Requested Things (and more)

### PR DESCRIPTION
**Pull request by Geevies**
* Slimes are now content after growing into adults or reproducing, meaning they won't attack humanoids for a while afterwards.
* The slime scanner has been tweaked to be a bit nicer.
* Slightly reduced the damage slimes do.
* Clicking a delivery chute with a held animal, such as a monkey, will now automatically deposit them into it.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -448,36 +448,41 @@ BREATH ANALYZER
...
+		to_chat(user, span("warning", "Warning: slime is hungry!"))
+	to_chat(user, span("notice", "Electric charge strength: [T.powerlevel]"))
+	to_chat(user, span("notice", "Growth progress: [T.amount_grown]/10"))
+	if(T.cores > 1)
+		to_chat(user, span("warning", "Anomalous number of slime cores detected."))
+	else if(!T.cores)
+		to_chat(user, span("warning", "No slime cores detected."))
+	to_chat(user,span( "notice", "Genetic Information:"))
+	if(T.slime_mutation[4] == T.colour)
+		to_chat(user, span("warning", "This slime does not evolve any further."))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/scanners.dm]:
```suggestion
		to_chat(user, span("warning", "This slime cannot evolve any further."))
```

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -448,36 +448,41 @@ BREATH ANALYZER
...
+				to_chat(user, span("notice", "Instability: [T.mutation_chance]% chance of mutation upon reproduction."))
 		else
-			user.show_message(text("Possible mutations: [], [], [], []", T.slime_mutation[1], T.slime_mutation[2], T.slime_mutation[3], T.slime_mutation[4]))
-			user.show_message("Genetic destability: [T.mutation_chance]% chance of mutation on splitting")
-	if (T.cores > 1)
-		user.show_message("Anomalious slime core amount detected")
-	user.show_message("Growth progress: [T.amount_grown]/10")
+			to_chat(user, span("notice", "Possible mutations: [T.slime_mutation[1]], [T.slime_mutation[2]], [T.slime_mutation[3]], [T.slime_mutation[4]]"))
+			to_chat(user, span("notice", "Instability: [T.mutation_chance]% chance of mutation upon reproduction."))
+	to_chat(user, span("notice", "**************************"))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/scanners.dm]: There has to be a way to do this with less copypaste.

Comment (Geevies) [reply code/game/objects/items/devices/scanners.dm]: Dunno. Don't think there is.

Comment (Geevies) [reply code/game/objects/items/devices/scanners.dm]: If ya know how, hit me up.

**Comment by LanceLynxx** [comment]
Eh, I think this makes working with slimes much too easy. At the very least don't change the damage values.

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -448,36 +448,41 @@ BREATH ANALYZER
...
+	if(T.slime_mutation[4] == T.colour)
+		to_chat(user, span("warning", "This slime cannot evolve any further."))
 	else
-		if (T.slime_mutation[3] == T.slime_mutation[4])
-			if (T.slime_mutation[2] == T.slime_mutation[1])
-				user.show_message(text("Possible mutation: []", T.slime_mutation[3]))
-				user.show_message("Genetic destability: [T.mutation_chance/2]% chance of mutation on splitting")
+		if(T.slime_mutation[3] == T.slime_mutation[4])
+			if(T.slime_mutation[2] == T.slime_mutation[1])
+				to_chat(user, span("notice", "Possible mutation: [T.slime_mutation[3]]"))
```
Comment (mikomyazaki) [inline code/game/objects/items/devices/scanners.dm]: Shouldn't this be listing two mutations? 1 and 3?

File: code/game/objects/items/devices/scanners.dm (hunk 2 above)
Comment (mikomyazaki) [reply code/game/objects/items/devices/scanners.dm]:
```var/list/poss_mutations = uniquelist(T.slime_mutation)
var/mutation_message = ""

for(var/i in 1 to poss_mutations.len)
	mutation_message += "[poss_mutations[i]]"
	if(i != poss_mutations.len)
		mutation_message += ", "
	else
		mutation_message += "."

to_chat(user, span("notice"), "Possible mutations: [mutation_message]")
to_chat(user, span("notice"), "Instability: [T.mutation_chance][poss_mutations.len == 2 ? "/2" : ""]% chance of mutation upon reproduction.")```

Something like that? (Written assuming my earlier comment about number of mutations error is correct)

File: code/modules/mob/living/carbon/slime/_mood.dm [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#define POUT "pout"
+#define SAD "sad"
+#define ANGRY "angry"
+#define MISCHIEVOUS "mischevous"
```
Comment (mikomyazaki) [inline code/modules/mob/living/carbon/slime/_mood.dm]:
```suggestion
#define MISCHIEVOUS "mischievous"
```

File: code/modules/mob/living/carbon/slime/powers.dm [hunk 5]
```diff
@@ -21,6 +21,8 @@
 		return "I cannot feed on other slimes..."
 	if(!Adjacent(M))
 		return "This subject is too far away..."
+	if(ishuman(M) && !istype(M, /mob/living/carbon/human/monkey) && content) // don't eat humans while content
```
Comment (mikomyazaki) [inline code/modules/mob/living/carbon/slime/powers.dm]:
```suggestion
	if(ishuman(M) && !isMonkey(M) && content) // don't eat humans while content
```

File: code/modules/mob/living/carbon/slime/powers.dm [hunk 6]
```diff
@@ -21,6 +21,8 @@
...
+		return "I'm already content..."
 	if(istype(M, /mob/living/carbon) && M.getCloneLoss() >= M.maxHealth * 2 || istype(M, /mob/living/simple_animal) && M.stat == DEAD)
 		return "This subject does not have an edible life energy..."
```
Comment (mikomyazaki) [inline code/modules/mob/living/carbon/slime/powers.dm]:
```suggestion
		return "This subject does not have any edible life energy..."
```

File: code/_onclick/other_mobs.dm [hunk 7]
```diff
@@ -87,8 +87,9 @@
 	setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
 
 	var/mob/living/M = A
-	if (istype(M))
-
+	if(ishuman(M) && !istype(M, /mob/living/carbon/human/monkey) && content)
```
[the remark this answers is not in the document]
Comment (Geevies) [reply code/_onclick/other_mobs.dm]: When I used isMonkey in testing, it didn't work, so I'm not gonna touch it right now.

File: code/_onclick/other_mobs.dm [hunk 8]
```diff
@@ -100,15 +101,21 @@
 						var/mob/living/carbon/human/H = M
 						stunprob *= H.species.siemens_coefficient
```
[the remark this answers is not in the document]
Comment (Geevies) [reply code/_onclick/other_mobs.dm]: Mmmmmaybe, probably not. I don't think it's too much of a problem to tweak it.

File: code/game/objects/items/devices/scanners.dm (hunk 3 above)
Comment (Geevies) [reply code/game/objects/items/devices/scanners.dm]: I genuinely have no idea, old code had it like this and no one's made an issue about it yet, so I assume it's working as intended.

File: code/modules/mob/living/carbon/slime/_mood.dm (hunk 4 above)
Comment (Geevies) [reply code/modules/mob/living/carbon/slime/_mood.dm]: this is the original typo, i don't want to scramble around in dmi's and stuff to fix it right now

File: code/modules/mob/living/carbon/slime/powers.dm (hunk 5 above)
Comment (Geevies) [reply code/modules/mob/living/carbon/slime/powers.dm]: see above

**Comment by Geevies** [comment]
The possmutation for loop is pretty interesting, but I think it makes the code more difficult to read.

File: code/game/objects/items/devices/scanners.dm (hunk 3 above)
Comment (mikomyazaki) [reply code/game/objects/items/devices/scanners.dm]: I think this is incorrect, but I'll fix it in a future PR of my own I guess.

**Pull request merged**